proof-systems/src/darlin/poly-commit/src/ipa_pc/mod.rs

```diff
@@ -30,12 +32,12 @@ mod tests;
 #[derive(Derivative)]
 #[derivative(Clone(bound = ""))]
 /// The inner product argument from [BCMS20](https://eprint.iacr.org/2020/499).
-pub struct InnerProductArgPC<G: Curve, D: Digest> {
+pub struct InnerProductArgPC<G: EndoMulCurve, FS: FiatShamirRng> {
```

Comment:
It's not doable in this way, as I need to explicitly implement IPACurve trait for the structs we have defined in algebra.
Implementing them in algebra, which is supposed to be generic, is not nice.
On the other hand, I cannot implement them here in poly-commit as it would throw an error for trying to implementing traits on foreign types.
The other way to do it is with macros, that however will make the development and debugging a bit bothersome.
As we discussed offline, let's leave like this for the moment